### ISSUE TITLE
V2 is currently dependent on V1

### ISSUE THREAD

**Issue by henryleduc** [nats-io/jwt#156]
As stated in the guide to major versioning for Go, you should replace all import paths from the original version to the v2 import path.
https://go.dev/blog/v2-go-modules

> Note that the v2 version is treated as a separate module from the v0 / v1 versions: both may coexist in the same build. So, if your v2+ module has multiple packages, you should update them to use the new /v2 import path: **otherwise, your v2+ module will depend on your v0 / v1 module.**

Not only is this bad practice but this also causes a problem, as there is no safe versions of the V1 dependency due to [CVE-2021-3127](https://nvd.nist.gov/vuln/detail/CVE-2021-3127).
So this means anybody using an internal proxy that quarantines dependencies with vulnerabilities is unable to use this package or any package dependent on it.

The only uses of the V1 packages in the V2 directory are for test files.
- [decoder_migration_test.go](https://github.com/nats-io/jwt/blob/master/v2/decoder_migration_test.go#L23)
- [genericclaims_test.go](https://github.com/nats-io/jwt/blob/master/v2/genericclaims_test.go#L22)

**Comment by wallyqs** [comment]
The major subdirectory approach although not common is one of the ok approaches as noted in the Go modules guide [here](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher), but agree that the tests should be refactored to avoid the dependency check fail (moving these tests into a `v2/test` folder should solve the issue I think in this case).

**Comment by henryleduc** [comment]
> The major subdirectory approach although not common is one of the ok approaches as noted in the Go modules guide [here](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher), but agree that the tests should be refactored to avoid the dependency check fail (moving these tests into a `v2/test` folder should solve the issue I think in this case).

I'm not sure moving the tests to a subdirectory would resolve the dependency check as the v1 package is required in the v2 go.mod

https://github.com/nats-io/jwt/blob/master/v2/go.mod#L4

**Comment by wallyqs** [comment]
Thanks, yes that `go.mod` file had to be updated as well.  Ok with the fixes in this branch (https://github.com/nats-io/jwt/commit/6e49afee39ac87a850114d6eb5f9a04f85b2c680) can separate things so that parts of the test dependency isolated just for testing, so now both of these would work:

```sh
# Test without external dependencies
go test github.com/nats-io/jwt/v2 -v

# Tests with external dependencies (v1 dep in this case)
go test -modfile=go_test.mod github.com/nats-io/jwt/v2/test -v
```

We may need to tweak the CI a bit to use the different modfile but this should make it so that the `jwt/v2` version can be imported without having to bring the v1 dependency and making the dependency check fail.
Thanks for the report!

**Comment by kozlovic** [comment]
Another approach is that, since the cve issue is fixed on both v1 and v2 (https://github.com/nats-io/jwt/pull/149/files), we could have simply released a new v1 version. We could do it even now.. the only thing is that it is going to be messy in term of release notes since likely it contains most of the things that have been done in the v2 releases, but would have to make sure.

**Comment by kozlovic** [comment]
We released v2.1.0 with the changed from PRs 157/158. Hope this helps: https://github.com/nats-io/jwt/releases/tag/v2.1.0

**Comment by henryleduc** [comment]
Thank you everyone for the speedy responses, fixes and releases 😄
